pybamm/models/full_battery_models/base_battery_model.py

```diff
@@ -633,6 +633,20 @@ def __init__(self, extra_options):
                     "be 'none': 'particle mechanics', 'loss of active material'"
                 )
 
+        if "true" in options["SEI on cracks"]:
+            sei_on_cr = options["SEI on cracks"]
+            p_mechanics = options["particle mechanics"]
+            if isinstance(p_mechanics, str) and isinstance(sei_on_cr, tuple):
+                p_mechanics = (p_mechanics, p_mechanics)
+            if any(
+                sei == "true" and mech != "swelling and cracking"
+                for mech, sei in zip(p_mechanics, sei_on_cr)
+            ):
+                raise pybamm.OptionError(
+                    "If 'SEI on cracks' is 'true' then 'particle mechanics' must be "
+                    "'swelling and cracking'."
```

Comment:
If swelling and cracking is required for "SEI on cracks", then can't we just enable swelling and cracking if the user wants SEI on cracks? That seems closer to the users intent

Reply:
The pattern you suggest is already implemented if a user *only* provides `SEI on cracks` in the options list, and not `particle mechanics`. However, in a scenario such as 
```
model = pybamm.lithium_ion.SPMe(
    {
        "particle mechanics": ("swelling only", "swelling only"),
        "SEI on cracks": "true",
    }
)
```
where both options are supplied, the users `particle mechanics` choice will override the default (where `swelling and cracking` was previously enabled based on the `SEI on cracks` value).

In a case where the user has specified both options, it seems unhelpful to assume which option is of more importance to the user by automatically overriding their mechanics choice. Throwing an error indicating an incompatible combination of options feels more appropriate to me - particularly given that this is such an easy fix for the user.